PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,463 @@
+      nonPrefiringProbMuonDownToken_(produces<double>("nonPrefiringProbMuonDown")),
+      nonPrefiringProbMuonUpSystToken_(produces<double>("nonPrefiringProbMuonSystUp")),
+      nonPrefiringProbMuonDownSystToken_(produces<double>("nonPrefiringProbMuonSystDown")),
+      nonPrefiringProbMuonUpStatToken_(produces<double>("nonPrefiringProbMuonStatUp")),
+      nonPrefiringProbMuonDownStatToken_(produces<double>("nonPrefiringProbMuonStatDown")),
+      dataeraEcal_(iConfig.getParameter<std::string>("DataEraECAL")),
+      dataeraMuon_(iConfig.getParameter<std::string>("DataEraMuon")),
+      useEMpt_(iConfig.getParameter<bool>("UseJetEMPt")),
+      doMuons_(iConfig.getParameter<bool>("DoMuons")),
+      prefiringRateSystUncEcal_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyECAL")),
+      prefiringRateSystUncMuon_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyMuon")),
+      jetMaxMuonFraction_(iConfig.getParameter<double>("JetMaxMuonFraction")) {
+  missingInputEcal_ = false;
+  missingInputMuon_ = false;
+
+  std::string fname = iConfig.getParameter<std::string>("L1Maps");
+  edm::FileInPath mapsfilepath("PhysicsTools/PatUtils/data/" + fname);
+  file_prefiringmaps_ = std::make_unique<TFile>(mapsfilepath.fullPath().c_str(), "read");
+  if (file_prefiringmaps_ == nullptr) {
+    missingInputEcal_ = true;
+    edm::LogError("L1PrefireWeightProducer")
+        << "File with maps not found. All prefiring weights set to 0. " << std::endl;
+  }
+  TString mapphotonfullname = "L1prefiring_photonptvseta_" + dataeraEcal_;
+  if (!file_prefiringmaps_->Get(mapphotonfullname)) {
+    missingInputEcal_ = true;
+    edm::LogError("L1PrefireWeightProducer")
+        << "Photon map not found. All photons prefiring weights set to 0. " << std::endl;
+  }
+  h_prefmap_photon_ = (TH2F*)file_prefiringmaps_->Get(mapphotonfullname);
+  TString mapjetfullname =
+      (useEMpt_) ? "L1prefiring_jetemptvseta_" + dataeraEcal_ : "L1prefiring_jetptvseta_" + dataeraEcal_;
+  if (!file_prefiringmaps_->Get(mapjetfullname)) {
+    missingInputEcal_ = true;
+    edm::LogError("L1PrefireWeightProducer") << "Jet map not found. All jets prefiring weights set to 0. " << std::endl;
+  }
+  h_prefmap_jet_ = (TH2F*)file_prefiringmaps_->Get(mapjetfullname);
+  file_prefiringmaps_->Close();
+
+  std::string fnameMuon = iConfig.getParameter<std::string>("L1MuonParametrizations");
+  edm::FileInPath paramsfilepath("PhysicsTools/PatUtils/data/" + fnameMuon);
+  file_prefiringparams_ = std::make_unique<TFile>(paramsfilepath.fullPath().c_str(), "read");
+  if (file_prefiringparams_ == nullptr) {
+    missingInputMuon_ = true;
+    edm::LogError("L1PrefireWeightProducer")
+        << "File with muon parametrizations not found. All prefiring weights set to 0." << std::endl;
+  }
+  TString paramName = "L1prefiring_muonparam_0.0To0.2_" + dataeraMuon_;
+  parametrization0p0To0p2_ = (TF1*)file_prefiringparams_->Get(paramName);
```

Comment:
Same here and below:
```suggestion
  parametrization0p0To0p2_ = file_prefiringparams_->Get<TF1>(paramName);
```
Is there any reason why you did not backported identically from the master these `Get` calls? A different version of root, maybe?

Reply:
I found that the Get calls don't work in 10_6_X, indeed.